server/server.js

```diff
@@ -163,7 +164,12 @@ APP.post('/api/campaign/loadAll', (req, res) => {
     const DB = CLIENT.db(DBNAME);
     DB.collection("campaigns").find().toArray(function(err, result) {
         if (err) throw err;
-        res.send(result);
+        let orderedResults = result.sort((a,b) => {
```

Comment:
if I am reading this correctly, you are sorting the results after they have been transferred from the database to the server. When we have a small number of campaigns, that's OK, but as the number of campaigns grows, this sorting will become more problematic, because it will use the server's memory and CPU. I would prefer if the sorting was done by the database. For efficiency of sorting, you would also need to add an database index on lastDonationTime.

Reply:
code updated. sort is happening within query